app/views/NotificationPrefView/index.js

```diff
@@ -0,0 +1,401 @@
+			}
+		}
+		if (Object.keys(params).length > 0) {
+			try {
+				await RocketChat.saveNotificationSettings(this.rid, params);
+			} catch (err) {
+				log('err_save_notification_settings', err);
+			}
+		}
+		this.room = JSON.parse(JSON.stringify(room));
+		EventEmitter.emit(LISTENER, { message: I18n.t('Settings_succesfully_changed') });
+	}
+
+
+	render() {
+		return (
+			<SafeAreaView style={sharedStyles.listSafeArea} testID='notificationPref-view'>
+				<StatusBar />
+				<ScrollView
+					{...scrollPersistTaps}
+					contentContainerStyle={sharedStyles.listContentContainer}
+					showsVerticalScrollIndicator={false}
+					testID='notificationPref-view-list'
+				>
+					<ListItem
+						title={I18n.t('Receive_Notification')}
+						testID='notificationPref-view-receive-notification'
+						right={() => this.renderSwitch('disableNotifications')}
+					/>
+					<Separator />
+					<Info info={I18n.t('Receive_notifications_from', { name: this.room.name })} />
+					<SectionSeparator />
+
+					<Separator />
+					<ListItem
+						title={I18n.t('Show_Unread_Counter')}
+						testID='notificationPref-view-unread_count'
+						right={() => this.renderSwitch('hideUnreadStatus')}
+					/>
+					<Separator />
+					<Info info={I18n.t('Show_Unread_Counter_Info')} />
+
+					<SectionSeparator />
+					<SectionTitle title={I18n.t('IN_APP_AND_DESKTOP')} />
+
+					<ListItem
+						title={I18n.t('Alert')}
+						testID='notificationPref-view-alert'
+						right={() => this.renderPicker('desktopNotifications')}
+					/>
+					<Separator />
+					<Info info={I18n.t('In_App_and_Desktop_Alert_info')} />
+
+					<SectionSeparator />
+					<SectionTitle title={I18n.t('Push_Notifications')} />
+
+					<ListItem
+						title={I18n.t('Alert')}
+						testID='notificationPref-view-push-notification'
+						right={() => this.renderPicker('mobilePushNotifications')}
+					/>
+					<Separator />
+					<Info info={I18n.t('Push_Notifications_Alert_Info')} />
+
+					<SectionSeparator />
+					<SectionTitle title={I18n.t('Desktop_Options')} />
+
+					<ListItem
+						title={I18n.t('Audio')}
+						testID='notificationPref-view-audio'
+						right={() => this.renderPicker('audioNotifications')}
+					/>
+					<Separator />
+					<ListItem
+						title={I18n.t('Sound')}
+						testID='notificationPref-view-sound'
+						right={() => this.renderPicker('audioNotificationValue')}
+					/>
+					<Separator />
+					<ListItem
+						title={I18n.t('Notification_Duration')}
+						testID='notificationPref-view-notification-duration'
+						right={() => this.renderPicker('desktopNotificationDuration')}
+					/>
+					<Separator />
+
+					<SectionSeparator />
+					<SectionTitle title={I18n.t('Email')} />
+
+					<ListItem
+						title={I18n.t('Alert')}
+						testID='notificationPref-view-email_alert'
+						right={() => this.renderPicker('emailNotifications')}
+					/>
+					<Separator />
+
+					<View
+						style={styles.container}
+					>
+						<TouchableOpacity
```

Comment:
Can you submit each option on change instead of having a `submit` button?
It's different than the rest of the app, but I think it's a better ux.

Reply:
Cool I'll try